state/factory/workingset.go

```diff
@@ -97,6 +97,11 @@ func (ws *workingSet) validate(ctx context.Context) error {
 			ws.height,
 		)
 	}
+	// Reject execution of chainID not equal the node's chainID
+	blkChainCtx := protocol.MustGetBlockchainCtx(ctx)
+	if blkChainCtx.ChainID != config.ChainID() {
```

Comment:
both config.ChainID() and blkChainCtx.ChainID comes from the genesis.Blockchain.ChainID, so this check is not effective